postgres.yml

```diff
@@ -6,8 +6,17 @@
     - "postgres_common_vars.yml"
 
   pre_tasks:
+    - name: "PostgreSQL | Get Postgres Master IP from DNS"
+      shell: "dig +short {{ postgres_host }}"
+      register: postgres_master_IP_dig
+    - name: "PostgreSQL | Set Postgres Master IP fact"
+      set_fact:
+        postgres_master: "{{ postgres_master_IP_dig.stdout }}"
     - include_vars: postgres_master_vars.yml
       when: "ansible_default_ipv4.address == postgres_master"
+    - name: PostgreSQL Master | Delete recovery.conf
+      file: path=/var/lib/postgresql/9.3/main/recovery.conf state=absent
+      when: "ansible_default_ipv4.address == postgres_master"
 
```

Comment:
@russss  mentioned that you should not need to remove recovery. When switching to master it would be automatically removed by postgres.